### ISSUE TITLE
Compilation error while using extension functions

### ISSUE THREAD

**Issue by bhavinmdesai** [tfcporciuncula/hilt-assisted-injection#1]
Hi,

I referred to your code when trying to do AssistedInjection into ViewModel and Hilt as a dependency injection framework. When I used your code, I am getting a compilation error. Can you please 'assist' here? :) 

![Screenshot 2023-08-23 at 5 24 20 PM](https://github.com/tfcporciuncula/hilt-assisted-injection/assets/6701463/cf89638d-d963-44ff-b3f1-d3dee8e5cdab)

**Comment by tfcporciuncula** [comment]
Yeah this code is pretty old and that API changed significantly starting from [lifecycle 2.5.0](https://developer.android.com/jetpack/androidx/releases/lifecycle#2.5.0). I just opened a PR updating everything so you can take a look at how the code should look like today: https://github.com/tfcporciuncula/hilt-assisted-injection/pull/2
